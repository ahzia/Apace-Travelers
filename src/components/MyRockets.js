import React from 'react';
import { useSelector } from 'react-redux';
import './ProfileMission.css';

const MyRockets = () => {
  const rockets = useSelector((state) => state.rocketReducer.rockets);
  const rocketsReserved = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <div>
      <h3 className="joined__mission__title">{rocketsReserved.length > 0 ? 'My Rockets' : 'No Rockets Reserved!'}</h3>
      <div className="joined__mission__container">
        {rocketsReserved.map((join) => (
          <div key={join.rocket_id}>
            <div>{join.reserved === true ? <p className="joined__mission__item border">{join.rocket_name}</p> : <p>{null}</p>}</div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default MyRockets;
