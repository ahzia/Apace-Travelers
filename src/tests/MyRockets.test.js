import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyRockets from '../components/MyRockets';

describe('Component test snapshot', () => {
  it('Rockets renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyRockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
