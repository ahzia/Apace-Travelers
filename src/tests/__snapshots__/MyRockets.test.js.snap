// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test snapshot Rockets renders correctly 1`] = `
<div>
  <h3
    className="joined__mission__title"
  >
    No Rockets Reserved!
  </h3>
  <div
    className="joined__mission__container"
  />
</div>
`;
